src/Symfony/Bundle/FrameworkBundle/Test/KernelShutdownOnTearDownTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use PHPUnit\Framework\TestCase;
+
+// Auto-adapt to PHPUnit 8 that added a `void` return-type to the tearDown method
+
+if (method_exists(\ReflectionMethod::class, 'hasReturnType') && (new \ReflectionMethod(TestCase::class, 'tearDown'))->hasReturnType()) {
+eval('
+    /**
+     * @internal
+     */
+    trait KernelShutdownOnTearDownTrait
+    {
+        protected function tearDown(): void
+        {
+            static::ensureKernelShutdown();
+        }
+    }
+');
```

Comment:
why the eval ?

Reply:
Had my answer in the comment https://github.com/symfony/symfony/pull/30124#discussion_r255305897